Module 6

IlyaBurlak opened the pull request.

IlyaBurlak committed:
init project

IlyaBurlak committed:
refactor: Настроил ESlint

IlyaBurlak committed:
fix: linter

IlyaBurlak committed:
Draft: минимальная версия для теста Redux

IlyaBurlak committed:
Draft: минимальная версия для теста Redux

IlyaBurlak committed:
refactor: улучшил инфу по слову

IlyaBurlak committed:
refactor: улучшил стили

IlyaBurlak committed:
refactor: сделал чистый нормальный поисковик

IlyaBurlak committed:
refactor: декомпозиция

IlyaBurlak committed:
refactor: декомпозиция

IlyaBurlak committed:
fix: Стилей + Интерфейсов

IlyaBurlak committed:
fix: .env

IlyaBurlak committed:
fix: мелкие правки

IlyaBurlak committed:
fix: exports

IlyaBurlak committed:
fix: мелкие правки

IlyaBurlak committed:
refactor: Исправил синтаксис

    Реализация была сложно читаемой

IlyaBurlak committed:
fix: Забытый catch

IlyaBurlak committed:
refactor: Исправил список при поиске

    Теперь при пустом input список слов очищается

IlyaBurlak committed:
refactor: Исправил Favorite Bug

  Была исправлена ошибка при которой поиск на странице Favorite дублировался в HomePage

IlyaBurlak committed:
refactor: глобальный переезд на uuid

   Был добавлен uuid и исправлена логика работы с запросами для оптимальной работы

IlyaBurlak committed:
fix: gitignore

IlyaBurlak committed:
Merge branch 'main' into module-6

tamtataaam commented on 6-module/src/components/WordList/WordList.tsx:
```diff
@@ -0,0 +1,82 @@
+      const found = Object.values(cache).find((v) => v?.meta?.uuid === uuid);
+      if (found) return found as SearchItem;
+      const word = (fav as any).word ?? '';
+      return { word, meta: { uuid } } as unknown as SearchItem;
```
as unknown as SearchItem - тоже плохая практика, об этом как раз рассказано в видео + не совсем понимаю для чего это здесь нужно

IlyaBurlak committed:
fix: Изменил названия переменных

IlyaBurlak committed:
refactor: Убрал any

IlyaBurlak committed:
fetch: Глобальна переделка Redux

   + Зацепил EsLint так как при сброке он собирал файлы d.ts и тд

IlyaBurlak committed:
refactor: Вынес логику Loading

IlyaBurlak committed:
refactor: Добавил selector + упростил компонент

IlyaBurlak committed:
refactor: Убрал inline типизацию

IlyaBurlak committed:
Merge remote-tracking branch 'origin/module-6' into module-6

tamtataaam commented on 6-module/src/features/words/utils/mappers.ts:
```diff
@@ -0,0 +1,32 @@
+import { Phonetic, Pronunciation, UroItem, WordItem } from '../../../types/word';
+import { cleanMerriamWebsterText } from '../../../utils/cleanMerriamWebsterText';
+
+export const mapPhonetics = (prs: Pronunciation[] | undefined): Phonetic[] => {
```
Вместо `(prs: Pronunciation[] | undefined)` можно написать покороче: `(prs?: Pronunciation[])`
Тоже самое и в остальных функциях

IlyaBurlak committed:
refactor: Общие замечания по коду

   Было сделано:
   1  Деструктуризация WordList
   2  Исправлены типы, убрана лишня типизация
   3  Добавлены новые хуки

IlyaBurlak committed:
fix: style + ESlint
